syncing/src/test/kotlin/maru/syncing/beaconchain/CLSyncServiceImplTest.kt

```diff
@@ -0,0 +1,493 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * This file is dual-licensed under either the MIT license or Apache License 2.0.
+ * See the LICENSE-MIT and LICENSE-APACHE files in the repository root for details.
+ *
+ * SPDX-License-Identifier: MIT OR Apache-2.0
+ */
+package maru.syncing.beaconchain
+
+import java.net.ServerSocket
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.random.Random
+import maru.config.P2P
+import maru.config.consensus.ElFork
+import maru.config.consensus.qbft.QbftConsensusConfig
+import maru.consensus.ConsensusConfig
+import maru.consensus.ForkIdHashProvider
+import maru.consensus.ForkIdHasher
+import maru.consensus.ForkSpec
+import maru.consensus.ForksSchedule
+import maru.consensus.StaticValidatorProvider
+import maru.consensus.qbft.DelayedQbftBlockCreator
+import maru.core.BeaconBlock
+import maru.core.BeaconBlockHeader
+import maru.core.BeaconState
+import maru.core.Seal
+import maru.core.SealedBeaconBlock
+import maru.core.Validator
+import maru.core.ext.DataGenerators
+import maru.core.ext.metrics.TestMetrics.TestMetricsFacade
+import maru.core.ext.metrics.TestMetrics.TestMetricsSystemAdapter
+import maru.crypto.Hashing
+import maru.database.BeaconChain
+import maru.database.InMemoryBeaconChain
+import maru.extensions.fromHexToByteArray
+import maru.p2p.P2PNetworkImpl
+import maru.p2p.PeerLookup
+import maru.p2p.messages.StatusMessageFactory
+import maru.serialization.ForkIdSerializers
+import maru.serialization.rlp.RLPSerializers
+import maru.syncing.beaconchain.pipeline.BeaconChainDownloadPipelineFactory.Config
+import net.consensys.linea.metrics.Counter
+import net.consensys.linea.metrics.MetricsFacade
+import org.apache.tuweni.bytes.Bytes32
+import org.assertj.core.api.Assertions.assertThat
+import org.awaitility.Awaitility.await
+import org.hyperledger.besu.crypto.KeyPair
+import org.hyperledger.besu.crypto.SignatureAlgorithm
+import org.hyperledger.besu.crypto.SignatureAlgorithmFactory
+import org.hyperledger.besu.ethereum.core.Util
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.times
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doAnswer
+import org.mockito.kotlin.never
+import org.mockito.kotlin.reset
+import org.mockito.kotlin.spy
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import tech.pegasys.teku.networking.p2p.libp2p.MultiaddrPeerAddress
+
+class CLSyncServiceImplTest {
```

Comment:
We're not using an argument to onSyncComplete handler anywhere, but to complete the coverage, we can introduce some assertions on the sync targets passed to the callback. Up to you